### ISSUE TITLE
'map' file not found error in ActiveModel.h. 

### ISSUE THREAD

**Issue by 696GrocuttT** [mousebird-consulting-inc/WhirlyGlobe#1406]
With v3.3 I'm getting "'map' file not found" error from the "#import <map>" statement in ActiveModel.h. I'm not exactly sure which point this broke, but it was working for me in v2.6.7.

This seems similar to this old issue https://github.com/mousebird-consulting-inc/WhirlyGlobe/issues/544, and like the old issue, I am pulling in WhirlyGlobe via a pod. I'm guessing it's something to do with the fact that WhirlyGlobe-umbrella.h is now pulling in ActiveModule.h, but I'm not sure how that file is constructed (I think its autogenerated?)
 
Thanks

Tom

**Comment by mousebird** [comment]
We have clients who use the pod so I poked through those projects.
To start with, one didn't have an umbrella and the other did.
I suspect the first one is just using the high level interface directly from Swift and didn't need it.
But for the second one we've got several "WhirlyGlobe-Maply-Umbrella.h" files.  They all just pull in WhirlyGlobeMaplyComponent.h.
ActiveModel.h is in the C++ part of the library and shouldn't be pulled into anything public.

So I'm curious what WhirlyGlobe-umbrella.h is and who's generating it.

**Comment by 696GrocuttT** [comment]
Just a bit of background. To pull in WhirlyGlobe from swift I'm just doing 
"import WhirlyGlobe"

It looks like this is pulling in Pods/Target Support Files/WhirlyGlobe/WhirlyGlobe.modulemap which just contains this:
```
framework module WhirlyGlobe {
  umbrella header "WhirlyGlobe-umbrella.h"

  export *
  module * { export * }
}
```
I'll have a go at pulling in WhirlyGlobe another way to avoid the framework.

**Comment by mousebird** [comment]
Do you need that import statement?  In looking at a couple of Swift-only client projects I don't see the imports and I thought the Pod xcodeproj took care of that.

Things get more complex if you're working in Obj-C.

**Comment by 696GrocuttT** [comment]
Unfortunately I get build errors if I comment out that import statement. My project is 99% swift, but there is the odd Obj-C file (but none of them use WhirlyGlobe)

I have found out a few interesting things though. It looks like both WhirlyGlobe.modulemap and WhirlyGlobe-umbrella.h are auto generated by CocoaPods based on the podspec. More specifically the umbrella header is based on "public_header_files" variable. It looks like ActiveModel.h got moved at some point and is now being picked up by the 'common/WhirlyGlobeLib/include/*.h' pattern in the public_header_files variable. There's some interesting background info here:
https://github.com/CocoaPods/CocoaPods/issues/6601
https://guides.cocoapods.org/syntax/podspec.html#public_header_files

While I *think* I understand why this is breaking, I've still no idea why this works for other people using pods+swift on v3.3
